src/dndbeyond/base/utils.js

```diff
@@ -264,8 +264,9 @@ async function buildAttackRoll(character, attack_source, name, description, prop
                     } else {
                         // Apply great weapon fighting to brutal damage dice
                         if ((character.hasClassFeature("Great Weapon Fighting", true) || character.hasFeat("Great Weapon Fighting", true)) &&
-                            properties["Attack Type"] == "Melee" &&
-                            ((properties["Properties"].includes("Versatile") && character.getSetting("versatile-choice") != "one") || properties["Properties"].includes("Two-Handed"))) {
+                            (properties["Attack Type"] == "Melee" &&
+                            ((properties["Properties"].includes("Versatile") && character.getSetting("versatile-choice") != "one") || properties["Properties"].includes("Two-Handed"))) ||
+                            name == "Polearm Master - Bonus Attack") {
```

Comment:
arg, it's still an `A && B || C` situation, but now it's `GWF && melee-and-stuff || polearm` . Would need to add parenthesis around the whole second part.

Reply:
And this is a good example of why I shouldn't code things when I can't pay attention... I think it's ok now...

Reply:
> And this is a good example of why I shouldn't code things when I can't pay attention... I think it's ok now...

Oh, I don't think it would have made any difference. It's really breaking my brain... 
In the end I simplified it to this line and it started to make sense :
`if (GWF && ((melee && ((versatile && not-one-handed) || two-handed)) || polearm)) {`

I think saving some of those as variables and doing a simpler if statement would have been quite helpful in this case.